code/game/machinery/_machinery.dm

```diff
@@ -180,10 +183,22 @@ Class Procs:
 
 /obj/machinery/emp_act(severity)
 	. = ..()
-	if(use_power && !stat && !(. & EMP_PROTECT_SELF))
+	if(use_power && !stat && !(. & EMP_PROTECT_SELF) && prob(100 / severity))
```

Comment:
I don't like random chance of emp

Reply:
```suggestion
	if(use_power && !stat && !(. & EMP_PROTECT_SELF))
```

Reply:
Im not removing the random chance of EMPs, it happens with almost all EMPs. A heavy EMP already has a 100% chance to disable a computer, so the 50% chance is only on the edge of the EMP range.

Reply:
ok fine, we have come to an agreement I will implement.
Light range will reduce time instead of having a random probability.